x/sequencer/proposal/process.go

```diff
@@ -29,24 +26,29 @@ var (
 
 func NewProcessProposalHandler(txConfig client.TxConfig, controller controller.ArweaveBlocksController, keeper *keeper.Keeper,
 	logger log.Logger) sdk.ProcessProposalHandler {
-	handler := &processProposalHandler{txConfig: txConfig, controller: controller, keeper: keeper, logger: logger}
+	blockValidator := newBlockValidator(keeper, controller, logger)
+	err := blockValidator.Start()
+	if err != nil {
+		panic(err)
+	}
+
+	handler := &processProposalHandler{txConfig, logger, blockValidator}
 	return handler.process
 }
 
 // Validates the block proposal for the presence and correctness of transactions with the Arweave block,
 // as well as the correctness of data items
 func (h *processProposalHandler) process(ctx sdk.Context, req abci.RequestProcessProposal) abci.ResponseProcessProposal {
 	now := time.Now()
-	h.initSortKeyForBlock(ctx)
-	for txIndex, txBytes := range req.Txs {
-		tx, err := h.txConfig.TxDecoder()(txBytes)
-		if err != nil {
-			h.rejectProposal("unable to decode the transaction", "err", err)
-			return rejectResponse
-		}
-		if !h.processProposalValidateTx(ctx, txIndex, tx) {
-			return rejectResponse
-		}
+	block := h.createBlock(ctx, req)
+	if block == nil {
+		return rejectResponse
+	}
+
+	h.blockValidator.Input <- block
```

Comment:
Make Input channel private, add a Send() method that sends to the channel and waits on Ctx.Done() to prevent hanging on writes here.

Reply:
Wait on Task's Ctx but also sdk.Context

Reply:
Validator should stop processing when sdk.Context is cancelled

Reply:
done